src/lib/ifaces/macsec.rs

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use netlink_packet_route::rtnl::link::nlas::InfoData;
+use netlink_packet_route::rtnl::link::nlas::InfoMacSec;
+use serde::{Deserialize, Serialize};
+
+const MACSEC_VALIDATE_DISABLED: u8 = 0;
+const MACSEC_VALIDATE_CHECK: u8 = 1;
+const MACSEC_VALIDATE_STRICT: u8 = 2;
+
+const MACSEC_OFFLOAD_OFF: u8 = 0;
+const MACSEC_OFFLOAD_PHY: u8 = 1;
+const MACSEC_OFFLOAD_MAC: u8 = 2;
+
+const MACSEC_CIPHER_ID_GCM_AES_128: u64 = 0x0080C20001000001;
+const MACSEC_CIPHER_ID_GCM_AES_256: u64 = 0x0080C20001000002;
+const MACSEC_CIPHER_ID_GCM_AES_XPN_128: u64 = 0x0080C20001000003;
+const MACSEC_CIPHER_ID_GCM_AES_XPN_256: u64 = 0x0080C20001000004;
+const MACSEC_DEFAULT_CIPHER_ID: u64 = 0x0080020001000001;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub enum MacSecValidate {
+    Disabled,
+    Check,
+    Strict,
+    Other(u8),
+    Unknown,
+}
+
+impl Default for MacSecValidate {
+    fn default() -> Self {
+        MacSecValidate::Unknown
+    }
+}
+
+impl From<u8> for MacSecValidate {
+    fn from(d: u8) -> Self {
+        match d {
+            MACSEC_VALIDATE_DISABLED => Self::Disabled,
+            MACSEC_VALIDATE_CHECK => Self::Check,
+            MACSEC_VALIDATE_STRICT => Self::Strict,
+            _ => Self::Other(d),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub enum MacSecOffload {
+    Off,
+    Phy,
+    Mac,
+    Other(u8),
+    Unknown,
+}
+
+impl Default for MacSecOffload {
+    fn default() -> Self {
+        MacSecOffload::Unknown
+    }
+}
+
+impl From<u8> for MacSecOffload {
+    fn from(d: u8) -> Self {
+        match d {
+            MACSEC_OFFLOAD_OFF => Self::Off,
+            MACSEC_OFFLOAD_PHY => Self::Phy,
+            MACSEC_OFFLOAD_MAC => Self::Mac,
+            _ => Self::Other(d),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "kebab-case")]
+#[non_exhaustive]
+pub enum MacSecCipherId {
+    DeprecatedGcmAes128,
+    GcmAes128,
+    GcmAes256,
+    GcmAesXpn128,
+    GcmAesXpn256,
+    Other(u64),
+    Unknown,
+}
+
+impl Default for MacSecCipherId {
+    fn default() -> Self {
+        MacSecCipherId::Unknown
```

Comment:
According to kernel the default is GcmAes128.

Reply:
Fixed, thanks!